### PR TITLE
Pull results through to post test write up page

### PR DESCRIPTION
**Pull request by ammarhaiderbjss**
[Link to story](https://jira.i-env.net/browse/MES-710)
- Refactors fault store to return an observable when requesting fault totals
- Uses test results observable for post test summary screen
- Adds text edit icon to faults in post test summary screen
- Adds a textboxModal component for adding write ups for faults on post test summary
- Adds `prepush` script to `package.json` with the test command

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "format:fix": "pretty-quick --staged",
     "format:all": "npx prettier --write \"src/**/*{.ts,.js,.json,.css,.scss}\"",
     "precommit": "run-s format:fix lint gitSecrets scanRepo",
```
Comment (ataylorBJSS) [inline package.json]: Maybe we should take gitSecrets and scanRepo out of precommit too?

Comment (zielinskikamil) [reply package.json]: hym, after our discussion that we had yesterday I agree with @ataylorBJSS  :)

File: src/components/post-test-summary-section/post-test-summary-section.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<ion-card class="mes-test-summary-section-card" [ngClass]="{'hidden': summary.total === 0}">
```
Comment (zielinskikamil) [inline src/components/post-test-summary-section/post-test-summary-section.html]: it's good but just fyi we can hide/show elements using `[hidden]="experssion" ` too :)

File: src/components/post-test-summary-section/post-test-summary-section.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  selector: 'post-test-summary-section',
+  templateUrl: 'post-test-summary-section.html'
+})
+export class PostTestSummarySectionComponent {
+  
+  @Input() summary: IFaultSummary;
+
+  faultTitleColourMap = [
+    { title: FaultTitle.Dangerous, colour: 'failRed' },
+    { title: FaultTitle.Serious, colour: 'seriousYellow' },
```
Comment (zielinskikamil) [inline src/components/post-test-summary-section/post-test-summary-section.ts]: name `seriousYellow` is awesome :)

File: src/providers/fault-store/fault-store.ts [hunk 4]
```diff
@@ -79,61 +79,77 @@ export class FaultStoreProvider {
...
+
+    return { dangerousFaultSummary, seriousFaultSummary, drivingFaultSummary }
 
   }
 
+  getFaultTotals(): Observable<ITestResults> {
+    return new Observable((observer: Observer<any>) => {
+      observer.next(this.calculateFaultTotals());
+      observer.complete();
+    });
```
Comment (danielchildsbjss) [inline src/providers/fault-store/fault-store.ts]: This works in the case we have here, but in the future if we rely on getFaultTotal to act as a dynamic observable I don't think this is going to work - Adding a new fault isn't going to re-trigger the calculation so your faultTotals are going to be wrong.
It is okay for now but if we rely on this for something like the running totals on the DL25 it won't work like we want it to.

**Pull request merged**